src/main/java/com/mitchej123/hodgepodge/mixins/early/minecraft/MixinGuiNewChat_CompactChat.java

```diff
@@ -34,23 +34,26 @@ public class MixinGuiNewChat_CompactChat {
     private boolean hodgepodge$deleteMessage;
 
     @Inject(method = "func_146237_a", at = @At("HEAD"))
-    private void hodgepodge$compactChat(IChatComponent imsg, int p_146237_2_, int p_146237_3_, boolean refresh,
+    private void hodgepodge$compactChat(IChatComponent imsg, int chatLineId, int updateCounter, boolean refresh,
             CallbackInfo ci) {
-        this.hodgepodge$deleteMessage = !refresh && ChatHandler.tryCompactMessage(imsg, this.chatLines);
+        this.hodgepodge$deleteMessage = !refresh && ChatHandler.tryCompactMessage(imsg, this.chatLines)
+                && chatLineId == 0;
```

Comment:
Check chatLineId before trying to compact the chat:
```suggestion
        this.hodgepodge$deleteMessage = !refresh && chatLineId == 0
                && ChatHandler.tryCompactMessage(imsg, this.chatLines);
```

Reply:
no, this is intentional, this way it will still append a number at the end of the line, but it will not delete the message